File: Array/3432. Count Partitions with Even Sum Difference.cpp
class Solution {
public:
    int countPartitions(vector<int>& nums) 
    {
        int ans = 0, left = 0, n = nums.size(); 
        int right = accumulate(nums.begin(), nums.end(), 0);
        for(int i = 0; i < n - 1; i++)
        {
            left += nums[i], right -= nums[i];
            ans += not (abs(right - left) % 2);
        }
        return ans;
    }
};

File: Prefix Sum/3432. Count Partitions with Even Sum Difference.py
class Solution:
    def countPartitions(self, nums: List[int]) -> int:
        ans = 0
        left_sum, right_sum = 0, sum(nums)
        for val in nums:
            left_sum += val
            right_sum -= val
            ans += not (left_sum -right_sum) % 2
        return max(0, ans - 1)